quil-py/quil/program/__init__.pyi

```diff
@@ -1,3 +1,70 @@
+"""
+The `quil.program` module contains classes for constructing and representing a Quil program.
+
+# Examples
+
+## Source Mapping for Calibration Expansion
+
+```py
+    import inspect
+
+    program_text = inspect.cleandoc(
+        \"\"\"
+        DEFCAL X 0:
+            Y 0
+
+        DEFCAL Y 0:
+            Z 0
+
+        X 0 # This instruction is index 0
+        Y 0 # This instruction is index 1
+        \"\"\"
+    )
+
+    # First, we parse the program and expand its calibrations
+    program = Program.parse(program_text)
+    expansion = program.expand_calibrations_with_source_map()
+    source_map = expansion.source_map()
+
+    # This is what we expect the expanded program to be. X and Y have been replaced by Z
+    expected_program_text = inspect.cleandoc(
+        \"\"\"
+        DEFCAL X 0:
+            Y 0
+
+        DEFCAL Y 0:
+            Z 0
+
+        Z 0 # This instruction is index 0
+        Z 0 # This instruction is index 1
+        \"\"\"
+    )
+
+    # In order to discover _which_ calibration led to each Z in the resulting program, we
+    # can interrogate the expansion source mapping:
```

Comment:
I like this section a lot, but I found the specific comments a bit confusing.  Perhaps we could revise this first comment to something like

"…can interrogate the expansion source mapping.  For instance, The X at index 0 should have been replaced with a Z at index 0.  Here's how we could confirm that: First, we…".

As it stands, the comment on line 46 made me think that we were also going to check the Y at index 1.  (I think adding that would probably be redundant.)

Another possibility would be to drop the ellipses on lines 51 and 63; they're not following on from a previous sentence.

Reply:
yep, fixed